src/main/java/teammates/common/datatransfer/AccountAttributes.java

```diff
@@ -65,6 +65,22 @@ public AccountAttributes(String googleId, String name, boolean isInstructor,
         this.studentProfile.googleId = this.googleId;
     }
     
+    /**
+     * Gets a deep copy of this object.
+     */
+    public AccountAttributes getCopy() {
+        boolean isStudentProfileNull = this.studentProfile == null;
+        if (isStudentProfileNull) {
+            this.studentProfile = new StudentProfileAttributes();
+        }
+        AccountAttributes copy = new AccountAttributes(this.toEntity());
+        if (isStudentProfileNull) {
```

Comment:
Hmm, why do we have to do so much just to get a copy? This looks confusing, if this is a limitation because of the way that we need to instantiate ```AccountAttributes```, I think a comment is necessary here so future maintainers will not be confused